tools/check_op_desc.py

```diff
@@ -0,0 +1,317 @@
+                    "The arg `{}` of Input '{}' is changed: from '{}' to `{}`.".
+                    format(arg, name, ori_value, new_value))
+
+        # 2. print outputs error message
+        Outputs_error = error_message.get(op_name, {}).get(OUTPUTS, {})
+        for name in Outputs_error.get(ADD, {}):
+            print("The added Output '{}' is not dispensable.".format(name))
+
+        for name in Outputs_error.get(DELETE, {}):
+            print("The Output '{}' is deleted.".format(name))
+
+        for name in Outputs_error.get(CHANGE, {}):
+            changed_args = Outputs_error.get(CHANGE, {}).get(name, {})
+            for arg in changed_args:
+                ori_value, new_value = changed_args.get(arg)
+                print(
+                    "The arg `{}` of Output '{}' is changed: from '{}' to `{}`.".
+                    format(arg, name, ori_value, new_value))
+
+        # 3. print attrs error message
+        attrs_error = error_message.get(op_name, {}).get(ATTRS, {})
+        for name in attrs_error.get(ADD, {}):
+            print("The added attr '{}' doesn't set default value.".format(name))
+
+        for name in attrs_error.get(DELETE, {}):
+            print("The attr '{}' is deleted.".format(name))
+
+        for name in attrs_error.get(CHANGE, {}):
+            changed_args = attrs_error.get(CHANGE, {}).get(name, {})
+            for arg in changed_args:
+                ori_value, new_value = changed_args.get(arg)
+                print(
+                    "The arg `{}` of attr '{}' is changed: from '{}' to `{}`.".
+                    format(arg, name, ori_value, new_value))
+
+
+if len(sys.argv) == 1:
+    '''
+    Print all ops desc in dict:
+    {op1_name:
+        {INPUTS:
+            {input_name1:
+                {DISPENSABLE: bool,
+                 INTERMEDIATE: bool,
+                 DUPLICABLE: bool
+                },
+            input_name2:{}    
+            },
+         OUTPUTS:{},
+         ATTRS:
+            {attr_name1:
+                {TYPE: int,
+                 GENERATED: bool,
+                 DEFAULT_VALUE: int/str/etc,
+                }
+            }    
+        }
+     op2_name:{}        
+    }
+    '''
+    all_op_protos_dict = get_all_ops_desc()
+    result = json.dumps(all_op_protos_dict)
+    print(result)
+elif len(sys.argv) == 3:
+    '''
+    Compare op_desc files generated by branch DEV and branch PR.
+    And print error message.
+    '''
+    with open(sys.argv[1], 'r') as f:
+        origin_op_desc = f.read()
+
+    with open(sys.argv[2], 'r') as f:
+        new_op_desc = f.read()
+
+    error_message = compare_op_desc(origin_op_desc, new_op_desc)
+    if error:
+        print_error_message(error_message)
+
+else:
+    print("Usage:\n" \
+          "\t1. python check_op_desc.py > OP_DESC_DEV.spec\n" \
+          "\t2. python check_op_desc.py > OP_DESC_PR.spec\n"\
+          "\t3. python check_op_desc.py > OP_DESC_DEV.spec OP_DESC_PR.spec > diff_message")
```

Comment:
Do you mean "python check_op_desc.py OP_DESC_DEV.spec OP_DESC_PR.spec > diff_message" ?

Reply:
Yes.  
Redundant `>` is deleted .